components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -409,6 +411,12 @@ private boolean isFlowHandlerInCurrentStepCanHandleRequest(AuthenticationContext
         return false;
     }
 
+    private boolean isIdentifierFirstRequestInitiatedFromMagicLink(AuthenticationContext context) {
+
+        return Boolean.TRUE.equals(
+                context.getProperty(IS_IDF_INITIATED_FROM_MAGIC_LINK_AUTH));
```

Comment:
We can use Boolean.parseBoolean(context.getProperty(IS_IDF_INITIATED_FROM_MAGIC_LINK_AUTH)) here